src/main/java/com/pusher/client/PusherOptions.java

```diff
@@ -196,6 +198,27 @@ public String buildUrl(final String apiKey) {
                 : wsPort, apiKey, URI_SUFFIX);
     }
 
+    /**
+     *
+     * The default value is Proxy.NO_PROXY.
+     *
+     * @param proxy
+     *            Specify a proxy, e.g. <code>options.setProxy( new Proxy( Proxy.Type.HTTP, new InetSocketAddress( "proxyaddress", 80 ) ) )</code>;
+     * @return this, for chaining
+     */
+    public PusherOptions setProxy(Proxy proxy){
+        this.proxy = proxy;
+        return this;
+    }
+
+    public Proxy getProxy() {
+        if (this.proxy == null) {
```

Comment:
Default the member variable to `Proxy.NOPROXY`. Never mutate or translate a value in a getter, it breaks all expectations that the caller will have.

This one breaks the relationship between the get and set.

```
options.getProxy(null);
Proxy x = options.getProxy();
if (x != null) System.out.println("Whut?!");
```